app/models/solr_document.rb

```diff
@@ -120,6 +120,20 @@ def file_ids
     self[:img_info] || self[:file_id] || []
   end
 
+  # Checks holdings to see if there are any physical holdings
+  # If there aren't and the library isn't sul, return the library name
+  def eresources_library_display_name
+    library = holdings.libraries&.first
```

Comment:
Couldn't a mhld cause problems here?  the Holding::Items are derived from `item_display_struct`, which is created from the combination of holding statements and items. Its possible to have holding statements and no items, which might cause a problem here.  I think it might be better to look at `SolrDocument#folio_items` so that you don't have to worry about holding statements.

Reply:
I think `holdings` is the right place? We're going down this path because these are e-resources without an associated item in FOLIO, just a holdings record.

Reply:
What about `folio_holdings` then?  I got the impression we're trying to move away from `item_display_struct`.   I'm also wondering if this method ought to move to OnlineComponent, as I don't see much use for this outside of that context.

Reply:
I am just adding this so I don't forget Monday. I tested folio_holdings when Justin first commented on it, it will work with `folio_holdings.first.effective_location.library`. 

Also I am wondering if the mhld is an issue would we need to update this bit of code as well? Or is this not a problem in ezproxy? 
https://github.com/sul-dlss/SearchWorks/blob/afd93f43ee03dfbab769fb67da19d1ccde619af1/app/models/links/ezproxy.rb#L55-L57

Reply:
Oh, yeah, that's probably something to look at too, but not for this PR.

Reply:
I've made https://github.com/sul-dlss/SearchWorks/pull/4028 for that one

Reply:
> I got the impression we're trying to move away from item_display_struct

I don't know where you got that impression. I don't think that's the case in general.

Reply:
Is the idea to keep two parallel data structures in Solr? I thought the idea was to migrate to the folio based one. Maybe this could be covered post standup next week. I don't think we have a shared vision of what we're working toward.